aya-obj/src/obj.rs

```diff
@@ -831,6 +840,179 @@ impl Object {
         Ok(())
     }
 
+    fn create_dummy_ksym_var(&mut self) -> Result<u32, ParseError> {
+        let btf = self.btf.as_mut().ok_or(ParseError::NoBTF)?;
+
+        let int_type_id = {
+            let mut found_id = None;
+            for (idx, t) in btf.types().enumerate() {
+                if let BtfType::Int(int) = t {
+                    if int.size == 4 {
+                        found_id = Some((idx) as u32);
+                        break;
+                    }
+                }
+            }
+            found_id
+        };
+
+        let int_type_id = if let Some(id) = int_type_id {
+            id
+        } else {
+            let name_offset = btf.add_string("int");
+            btf.add_type(BtfType::Int(crate::btf::Int::new(
+                name_offset,
+                4,
+                crate::btf::IntEncoding::Signed,
+                0,
+            )))
+        };
+
+        debug!("Found/created int type_id: {}", int_type_id);
+        if let Ok(BtfType::Int(int)) = btf.type_by_id(int_type_id) {
+            debug!(
+                "Int type size: {}, encoding: {:?}",
+                int.size,
+                int.encoding()
+            );
+        }
+
+        let name_offset = btf.add_string("dummy_ksym");
+        let dummy_var_id = btf.add_type(BtfType::Var(crate::btf::Var::new(
+            name_offset,
+            int_type_id,
+            crate::btf::VarLinkage::Global,
+        )));
+
+        debug!("Created dummy_var type_id: {}", dummy_var_id);
+        if let Ok(BtfType::Var(var)) = btf.type_by_id(dummy_var_id) {
+            debug!("Dummy var points to type_id: {}", var.btf_type);
+        }
+
+        Ok(dummy_var_id)
+    }
+
+    /// Collects extern kernel symbols from BTF datasec entries.
+    fn collect_ksyms_from_btf(&mut self) -> Result<(), ParseError> {
```

Comment:
I think same here - this method is collecting ksyms from program's BTF. It could belong to `Btf` instead of `Obj`...